hoomd/md/methods.py

```diff
@@ -580,6 +583,144 @@ def _attach(self):
         # Attach param_dict and typeparam_dict
         super()._attach()
 
+class NVE_Rattle(_Method):
```

Comment:
I'm not a fan of the `_Rattle` suffix, but I don't have any better suggestions right now. Anyone have a better idea? @b-butler

Reply:
My immediate response to @joaander's comment was pretty much exactly what @b-butler suggested, I almost typed it up before I realized that. Yes, I think the simple solution is to rename `manifold` parameter to `constraint_manifold` so that it's very clear that it's used to constrain dynamics, then internally check `if constraint_manifold is None` to determine whether to use the Rattle integrator or not.